lib/services/hpcpackage_remain.pm

```diff
@@ -53,17 +51,17 @@ sub compare_pkg {
     $list1 //= '';
     $list2 //= '';
     # Make a list of installed packages and diff against list from installed packages
-    @diffpkg = split('\n', script_output("diff $list1 $list2 | grep -E '^>' | cut -d' ' -f2 |  grep -E '.*-hpc.*' |   grep -vE 'system|module|suse' |  grep -E '.*_[[:digit:]]+_[[:digit:]]+.*-gnu|.*_[[:digit:]]+_[[:digit:]]+.*-hpc' |  grep -vE '.*-static\$' | tee /tmp/diffpkg.txt", proceed_on_failure => 1, timeout => 180));
+    my @diffpkg = split('\n', script_output("diff $list1 $list2 | grep -E '^>' | cut -d' ' -f2 |  grep -E '.*-hpc.*' |   grep -vE 'system|module|suse' |  grep -E '.*_[[:digit:]]+_[[:digit:]]+.*-gnu|.*_[[:digit:]]+_[[:digit:]]+.*-hpc' |  grep -vE '.*-static\$' | tee /tmp/diffpkg.txt", proceed_on_failure => 1, timeout => 180));
 }
 
 # Confirm all packages in the list generated remain installed
 sub check_pkg {
-    my @pkglist = split('\n', script_output("rpm -qa", proceed_on_failure => 1, timeout => 180));
-    my %hash_a  = map { $_ => 1 } @pkglist;
-    my %hash_b  = map { $_ => 1 } @diffpkg;
-    diag 'After: ' . Dumper(\@pkglist);
-    diag @diffpkg;
-    my @b_only = grep { !$hash_a{$_} } @diffpkg;
+    my $arch    = get_var('ARCH');
+    my @pkglist = split('\n', script_output("rpm -qa | tee /tmp/afterpkg.txt", proceed_on_failure => 1, timeout => 180));
+    my $before  = '/tmp/diffpkg.txt';
+    my $after   = '/tmp/afterpkg.txt';
+    my $out     = script_output("diff $before $after | grep -iE '*.-[[:digit:].]$arch'", proceed_on_failure => 1, timeout => 180);
```

Comment:
this only make filter on those diff string with digital suffix, but not Fuzzy match